gocourse07/birdaviary/cmd/main.go

```diff
@@ -0,0 +1,52 @@
+/*
+Вам потрібно створити програму для управління вольєром для екзотичних птахів у розумному зоопарку. Вольєр має сенсори,
+які вимірюють температуру, яскравість освітлення та вологість, і вони передають ці дані до центральної системи, яка,
+в свою чергу, зберігає їх у памʼять (така от база даних). Раз на добу сенсори відключаються для технічної перевірки,
+але потім знову продовжують працювати. Центральна система також перезавантажується раз на добу.
+Робота кожного сенсора — окрема горутина. Коли сенсори відключаються, горутини безпечно вимикається. Те саме з
+центральною системою. Процес запису в памʼять повинен бути (штучно) тривалим. І у випадку, коли центральна система
+планово відключається, вона це мусить робити лише після того, як всі записи в базу виконались.
+*/
+package main
+
+import (
+	"os"
+	"sync"
+
+	"github.com/sarff/prjctr-golang_beginning/gocourse07/birdaviary/internal/centralsystem"
+	"github.com/sarff/prjctr-golang_beginning/gocourse07/birdaviary/internal/logger"
+	"github.com/sarff/prjctr-golang_beginning/gocourse07/birdaviary/internal/sensor"
+	"github.com/sarff/prjctr-golang_beginning/gocourse07/birdaviary/internal/storage"
+)
+
+func main() {
+	wg := new(sync.WaitGroup)
+	log := logger.New(os.Stdout)
+	db := storage.NewStorage(log)
+	cs := centralsystem.NewCentralSystem(db, log)
+
+	var tempSensor, humiditySensor, brightnessSensor sensor.Sensor
+
+	tempSensor = sensor.NewSensor(cs, log, "TemperatureSensor")
+	humiditySensor = sensor.NewSensor(cs, log, "HumiditySensor")
+	brightnessSensor = sensor.NewSensor(cs, log, "BrightnessSensor")
```

Comment:
В конкретній імплементації немає якогось бенефіту в використанні інтерфейса `sensor.Sensor`. Значно простіше та менше коду ось так: 
```suggestion
	tempSensor := sensor.NewSensor(cs, log, "TemperatureSensor")
	humiditySensor := sensor.NewSensor(cs, log, "HumiditySensor")
	brightnessSensor := sensor.NewSensor(cs, log, "BrightnessSensor")
```

Reply:
Бенефіт був би, якщо задекларувати колекцію сенсорів та працювати з ними однаково:

```go
	sensors := []sensor.Sensor{
		sensor.NewTemperatureSensor(cs, log),
		sensor.NewHumiditySensor(cs, log),
		sensor.NewBrightnessSensor(cs, log),
	}
	wg.Add(len(sensors))

	for _, s := range sensors {
		go s.Start(wg)
	}
```